# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: mycobot_280_gazebo
  joint_model_group_pose: HOME

generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - joint2_to_joint1
    - joint3_to_joint2
    - joint4_to_joint3
    - joint5_to_joint4
    - joint6_to_joint5
    - joint6output_to_joint6
  sim_control_mode: 0  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# PID controller settings for each joint
gazebo_ros_control/pid_gains:
  joint2_to_joint1: {p: 100.0, i: 0.01, d: 10.0}
  joint3_to_joint2: {p: 100.0, i: 0.01, d: 10.0}
  joint4_to_joint3: {p: 100.0, i: 0.01, d: 10.0}
  joint5_to_joint4: {p: 100.0, i: 0.01, d: 10.0}
  joint6_to_joint5: {p: 100.0, i: 0.01, d: 10.0}
  joint6output_to_joint6: {p: 100.0, i: 0.01, d: 10.0}

mycobot_position_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - joint2_to_joint1
    - joint3_to_joint2
    - joint4_to_joint3
    - joint5_to_joint4
    - joint6_to_joint5
    - joint6output_to_joint6

controller_list:
  - name: mycobot_position_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - joint2_to_joint1
      - joint3_to_joint2
      - joint4_to_joint3
      - joint5_to_joint4
      - joint6_to_joint5
      - joint6output_to_joint6
